pkg/command/permission_service.go

```diff
@@ -0,0 +1,165 @@
+package command
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/arquillian/ike-prow-plugins/pkg/github"
+)
+
+// PermissionService keeps user name and PR loader and provides information about the user's permissions
+type PermissionService struct {
+	Client   *github.Client
+	User     string
+	PRLoader *github.PullRequestLazyLoader
+}
+
+// NewPermissionService creates a new instance of PermissionService with the given client, user and pr loader
+func NewPermissionService(client *github.Client, user string, prLoader *github.PullRequestLazyLoader) *PermissionService {
+	return &PermissionService{
+		Client:   client,
+		User:     user,
+		PRLoader: prLoader,
+	}
+}
+
+// PermissionStatus keeps information about the user, his permissions and which roles are approved and which rejected
+type PermissionStatus struct {
+	User           string
+	UserIsApproved bool
+	ApprovedRoles  []string
+	RejectedRoles  []string
+}
+
+func (s *PermissionService) newPermissionStatus(allowedRoles ...string) *PermissionStatus {
+	return &PermissionStatus{User: s.User, ApprovedRoles: allowedRoles}
+}
+
+func (s *PermissionStatus) reject() *PermissionStatus {
+	s.UserIsApproved = false
+	return s
+}
+
+func (s *PermissionStatus) allow() *PermissionStatus {
+	s.UserIsApproved = true
+	return s
+}
+
+func (s *PermissionStatus) constructMessage(operation, command string) string {
+	msg := fmt.Sprintf(
+		"@%s has %s a command `%s` but this user is not allowed to do that. "+
```

Comment:
> but this user is not allowed to do that. 

how about

> but this will not have any effect due to insufficient permission

Plus maybe log users' permission too?

Reply:
sure, that sounds better

Reply:
implemented/changed in 20d461e